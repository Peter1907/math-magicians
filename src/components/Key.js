import React from 'react';
import PropTypes from 'prop-types';

class Key extends React.Component {
  keyPress = (e) => {
    const { keyPress } = this.props;
    keyPress(e.target.textContent);
  }

  render() {
    const { classes, name } = this.props;
    return (
      <button type="button" className={classes} onClick={this.keyPress}>{name}</button>
    );
  }
}

Key.propTypes = {
  keyPress: PropTypes.func.isRequired,
  classes: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default Key;
