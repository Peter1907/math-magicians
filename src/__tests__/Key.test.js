/* eslint-disable*/

import renderer from 'react-test-renderer';
import Key from '../components/Key';

it('renders correctly', () => {
  const state = {
    total: null,
    next: null,
    operation: null,
  }
  const setState = () => {
    state = {
      total: this.props.name,
      next: null,
      operation: null,
    }
  }
  const tree = renderer
    .create(<Key classes="key" name="7" state={state} setState={setState} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
