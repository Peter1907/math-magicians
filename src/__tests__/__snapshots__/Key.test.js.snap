// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="key"
  onClick={[Function]}
  type="button"
>
  7
</button>
`;
